src/ExportMap.js

```diff
@@ -497,6 +497,28 @@ ExportMap.parse = function (path, content, context) {
     m.reexports.set(s.exported.name, { local, getImport: () => resolveImport(nsource) });
   }
 
+  function captureDependencyWithSpecifiers(n) {
+    // import type { Foo } (TS and Flow)
+    const declarationIsType = n.importKind === 'type';
+    // import './foo' or import {} from './foo' (both 0 specifiers) is a side effect and
+    // shouldn't be considered to be just importing types
+    let specifiersOnlyImportingTypes = n.specifiers.length;
+    const importedSpecifiers = new Set();
+    n.specifiers.forEach(specifier => {
+      if (supportedImportTypes.has(specifier.type)) {
+        importedSpecifiers.add(specifier.type);
+      }
+      if (specifier.type === 'ImportSpecifier') {
```

Comment:
should the more specific check come first, and `else if` to the more general check?